Topic #40 - Adding Contribution guidelines and reference in README

james-grant1 opened the pull request:
Fixes #40

james-grant committed:
Initial CONTRIBUTING borrowed from another project

james-grant committed:
Initial migration of CONTRIBUTING to Isambard docs

james-grant committed:
Added reference to CONTIRUBTING in README

james-grant committed:
Fixed couple of typos, omissions

christopheredsall commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,100 @@
+# Contributing to Intro to CPP
```
What is CPP?

james-grant1 replied:
Relic from source, updated to Isambard Docs

christopheredsall commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,100 @@
+### Git Commit Messages
+
+* Title should briefly describe changes - use multi-line where necessary
+* Should be imperative
+* Typically use past tense
```
Not suggesting a change, because, as the document says, these are guidelines not rules. But aren't these two items in contradiction (in English)?

I prefer the imperative, e.g.
> - Fix a couple of typos, omissions
> - Add reference to CONTRIBUTING in README

c.f.

> - Fixed couple of typos, omissions 
> - Added reference to CONTRIBUTING in README

Because from my point of view when reading the log the question I ask is "what does this change do?" rather than "what has this developer done?"

Anyway, that's a matter of taste and I think I might be outvoted on GitHub

james-grant1 replied:
It does seem a little contradictory, not sure whether that was in the original or I've edited badly, remove past tense reference.

christopheredsall commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,100 @@
+
+### Documenting
+
+When adding to documentation, use reStructuredText format `rst`.
```
Maybe link to reStructuredText site, e.g. the quickstart
```
... use [reStructuredText](https://docutils.sourceforge.io/docs/user/rst/quickstart.html) format ...
```

james-grant1 replied:
This is linked above but no reason not to have it here too.

christopheredsall reviewed (changes requested):
Thanks for that James, it will be useful.

Can I get you to change the title? After that it is mergable,

Add a link to the reStructuredText site at your discression.

james-grant committed:
Updates for cedsall PR comments

Updated title to Isambard docs
Removed reference to commit messages in past tense
Added link to rst format

christopheredsall reviewed (commented):
+1 LGTM
